frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/RenameBackgroundable.java

```diff
@@ -78,6 +84,57 @@ public void run(ProgressIndicator indicator) {
     // (Stepping deeper is not an option since we would lose some important logic or become very dependent on the internals of git4idea).
     // Hence, we wait for the creation of the branch (with exponential backoff).
     waitForCreationOfLocalBranch(gitRepository, newBranchName);
+
+    if (gitNewBranchOptions.shouldSetTracking()) {
+      val remote = currentBranchSnapshot.getRemoteTrackingBranch();
+      assert remote != null : "shouldSetTracking is true but the remote branch does not exist";
+      handleSetTrackingBranch(remote.getRemoteName());
+    }
+  }
+
+  @UIThreadUnsafe
+  private void handleSetTrackingBranch(String remoteName) {
+    val newBranchName = gitNewBranchOptions.getName();
+    val renamedRemoteName = "${remoteName}/${newBranchName}";
+
+    // setUpstream: git branch --set-upstream-to <upstreamBranchName> <branchName>
+    val setUpstream = Git.getInstance().setUpstream(gitRepository, renamedRemoteName, newBranchName);
+    var error = setUpstream.getErrorOutputAsJoinedString();
+
+    if (!setUpstream.success()) {
+      if (setUpstream.getErrorOutput().get(0)
```

Comment:
Whoops I'd be careful about the case with no error output at all :frown:

Reply:
In case `git branch` gets killed or whatnot and doesn't produce any stderr in time